File: server/src/game/effects/HealEffect.ts
import { Player } from '../Player.js';
import type { ICardEffect } from './ICardEffect.js';
import type { EffectContext } from './EffectContext.js';

export class HealEffect implements ICardEffect {
  constructor(public readonly amount: number) {}

  execute(context: EffectContext): void {
    const target = context.target;
    if (target instanceof Player) {
      target.heroState.heal(this.amount);
    }
  }
}

File: server-java/src/main/java/vn/coincard/server/game/effects/HealEffect.java
package vn.coincard.server.game.effects;

import vn.coincard.server.game.Minion;
import vn.coincard.server.game.Hero;

public class HealEffect implements CardEffect {
  private final int amount;

  public HealEffect(int amount) { this.amount = amount; }

  @Override
  public void execute(EffectContext context) {
    if (context.target instanceof Hero hero) {
      hero.heal(amount);
    } else if (context.target instanceof Minion m) {
      m.heal(amount);
    }
  }
}
